src/cljs/main/broadfcui/common.cljs

```diff
@@ -246,11 +246,11 @@
 (def login-scopes ["email" "profile"])
 (def storage-scopes (conj login-scopes "https://www.googleapis.com/auth/devstorage.read_only"))
 
-(defn has-terra-return? []
-  (string/includes? js/window.location.search "return=terra"))
+(def return-app (js* "new URLSearchParams(document.location.search).get('return')"))
 
-(defn has-firecloud-return? []
-  (string/includes? js/window.location.search "return=firecloud"))
+(defn make-return-url [app]
+  (string/replace (config/terra-base-url) "*" app))
 
-(defn has-return? []
-  (or (has-terra-return?) (has-firecloud-return?)))
+(defn get-return-url [] (make-return-url (if (= return-app "terra") "app" return-app)))
```

Comment:
two comments:

* if this is ever called outside of a `(when common/has-return? ...)` clause, it will return something like "null.terra.bio", right? Should probably handle the `null` case,  something like:
```
(defn get-return-url [] (make-return-url (if (or (nil? return-app) (= return-app "terra")) "app" return-app)))
```
* More fundamentally, this function encodes the business logic of how we use hostnames - the "terra"="app"  logic is hardcoded. I don't know how many more of these we plan to release, but how about something a little more future-flexible, logic that does:
  1.  if `@config` contains a value for `(str return-app "TerraUrl")`, use the value from config
  2. else, replace `*` in the terra-base-url with the value of `return-app`

This way, we can handle the business-logic exceptions by adding config values, instead of changing this code.

Reply:
I think, by design, base terra is very much the only exception. On the terra-ui side, it's coded as: `export const returnParam = () => getAppName().toLowerCase()`, and for anything else, we're intentionally keeping the name and the url the same. If anything, I'd rather codify that pairing further.